numpy/distutils/ccompiler.py

```diff
@@ -496,6 +496,18 @@ def CCompiler_customize(self, dist, need_cxx=0):
     # See FCompiler.customize for suggested usage.
     log.info('customize %s' % (self.__class__.__name__))
     customize_compiler(self)
+
+    try:
```

Comment:
It seems to me like trying to strip out flags that are set already may be confusing. This would also filter out flags explicitly set in for example `CFLAGS` right?

Reply:
Oh my bad, I fixed it by execluding debug flags only if they weren't part of env vars.